core/src/abi.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{ConsensusError, ConsensusState, ensure};
+
+// Defines the Solidity struct layouts for public types
+alloy_sol_types::sol! {
+    struct Checkpoint {
+        uint64 epoch;
+        bytes32 root;
+    }
+
+    struct State {
+        Checkpoint current_justified;
+        Checkpoint finalized;
+    }
+
+    struct Journal {
+        State pre_state;
+        State post_state;
+        uint64 finalized_slot;
```

Comment:
I am still considering whether we should add the execution block hash or the execution state root. These might be useful for certain applications.